### ISSUE TITLE
Unable to resolve the filesystem mount because the path (dir/file.jpg) is missing a separator (://)

### ISSUE THREAD

**Issue by jmontoyaa** [dustin10/VichUploaderBundle#1190]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | yes
| Bundle version  | dev-master/1.17 
| Symfony version | 5.2.5
| PHP version     | 8.0

#### Summary

VichUploader doen't add the protocol part when getting a file when using the oneup_flysystem.
The MountManager in OneUP doesn't know how to find the file and fires the "missing a separator" (://) exception.

By "protocol part" I mean the text before ://

`oneup_flysystem.product_image_fs_filesystem://filename.txt`

VichUploader used to add that part, but it was removed when migrating to OneUp version 4 here:

https://github.com/dustin10/VichUploaderBundle/commit/83658542fba403fcca30fc303e30f8462a821811

#### Current behavior

The exception League\Flysystem\UnableToResolveFilesystemMount  appears when trying to upload an object.

#### How to reproduce

I modified a test here:

https://github.com/dustin10/VichUploaderBundle/pull/1189

And it fails as expected here:

https://github.com/dustin10/VichUploaderBundle/actions/runs/683264188

The line: 

https://github.com/dustin10/VichUploaderBundle/blob/master/tests/VichUploaderBundleTest.php#L121

Is very convenient for the test :) because it already gives the the right answer with the protocol part, but in fact it should be "filename.txt".

#### Expected behavior

Upload file correctly.

**Comment by garak** [comment]
I have to ask an advice from @fullbl, since I don't know anything about Flysystem

**Comment by fullbl** [comment]
Hi!

Flysystem 1.* had this MountManager, as documented here: https://flysystem.thephpleague.com/v1/docs/advanced/mount-manager/, but I can't find any documentation for the same in Flysystem 2, tough the class still exists.

What it does is to connect two different filesystems, so you can read from one and write to another.

How is it supposed to work? Should it write to first or to second filesystem? 

Here I see two possibilities:
- disallow the usage of MountManagers (and remove it from the documentation)
- ask the user which filesystem to use when uploading files

What do you think?

**Comment by jmontoyaa** [comment]
Yes, in Flysystem v2 there are some issues with MountManager. Example: https://github.com/thephpleague/flysystem/pull/1263 

I agree with your suggestions, I added a fix in order to use the correct oneup filesystem

https://github.com/dustin10/VichUploaderBundle/pull/1189
